chaco/data_frame_plot_data.py

```diff
@@ -0,0 +1,180 @@
+""" Defines DataFramePlotData.
+"""
+
+# Enthought library imports
+from traits.api import Bool, Instance, Property
+
+# Local, relative imports
+from .abstract_plot_data import AbstractPlotData
+
+
+class DataFramePlotData(AbstractPlotData):
+    """ A PlotData implementation class that handles a DataFrame.
+
+    By default, it doesn't allow its input data to be modified by downstream
+    Chaco components or interactors. The index is availble as data unless there
```

Comment:
I cannot spell that word right. I need to either stop using it or spell check every time I do. :D